src/tools/sfdk/sdkmanager.cpp

```diff
@@ -1348,15 +1348,37 @@ bool SdkManager::mapEnginePaths(QString *program, QStringList *arguments, QStrin
         {cleanSharedSrc, Constants::BUILD_ENGINE_SHARED_SRC_MOUNT_POINT, caseInsensitiveOnWindows}
 #endif
     };
+    *program = QDir::fromNativeSeparators(*program);
+
     for (const Mapping &mapping : mappings) {
         qCDebug(sfdk) << "Mapping" << mapping.hostPath << "as" << mapping.enginePath;
         program->replace(mapping.hostPath, mapping.enginePath, mapping.cs);
-        arguments->replaceInStrings(mapping.hostPath, mapping.enginePath, mapping.cs);
+        QRegularExpression::PatternOption reCaseSensitivity = (mapping.cs == Qt::CaseInsensitive)
+                ? QRegularExpression::CaseInsensitiveOption : QRegularExpression::NoPatternOption;
+        QRegularExpression re(QString("(?:%1|%2)([^:;,'\"]*)")
+                              .arg(QRegularExpression::escape(mapping.hostPath))
+                              .arg(QRegularExpression::escape(QDir::toNativeSeparators(mapping.hostPath))),
+                              reCaseSensitivity);
+        auto map = [=](QString string) {
+            QRegularExpressionMatchIterator reIterator = re.globalMatch(string);
+            QMap<QString,QString> replacements;
+            while (reIterator.hasNext()) {
+                QRegularExpressionMatch match = reIterator.next();
+                QString fullNativeHostPath = match.captured(0);
+                QString fullEnginePath = QDir::fromNativeSeparators(mapping.enginePath + match.captured(1));
+                replacements.insert(fullNativeHostPath, fullEnginePath);
+            }
+            for (auto i = replacements.end(); i != replacements.begin();) {
+                --i;
+                string.replace(i.key(), i.value());
+            }
```

Comment:
const_iterator (cend, cbegin) would be a little better. While at it, I would also consider using std::map instead of QMap and iterating with const_reverse_iterator, so it could be compactized as

```suggestion
            for (auto i = replacements.crbegin(); i != replacements.crend(); ++i)
                string.replace(i.key(), i.value());
```

Reply:
done